depmapomics/fingerprinting.py

```diff
@@ -0,0 +1,357 @@
+            n.append(
+                tuple(
+                    ref.loc[
+                        j,
+                        [
+                            "arxspan_id",
+                            "version",
+                            "datatype",
+                            "participant_id",
+                            "stripped_cell_line_name",
+                        ],
+                    ].values
+                )
+            )
+        previ = i
+    return shouldnt
+
+
+def add_sample_batch_pairs(wm, working_dir=WORKING_DIR):
+    # add and update sample_batch_pairs and sample_batch_pair_set
+    all_sample_sets = wm.get_entities("sample_set").index
+    sample_set_a_list = []
+    sample_set_b_list = []
+    pair_ids = []
+    for s in all_sample_sets:
+        for t in all_sample_sets:
+            sample_set_a_list.append(s)
+            sample_set_b_list.append(t)
+            pair_ids.append(s + "-" + t)
+
+    # pair_df contains all possible pairs between all sample_sets
+    pair_df = pd.DataFrame(
+        np.array([sample_set_a_list, sample_set_b_list]).T,
+        columns=["sample_batch_a", "sample_batch_b"],
+        index=pair_ids,
+    )
+    pair_df.index.name = "entity:sample_batch_pair_id"
+
+    # update sample_batch_pair
+    try:
+        wm.upload_entities("sample_batch_pair", pair_df)
+    except:
+        print("still can't update sample_batch_pair")
+        # in case it does not work
+        pair_df.to_csv(working_dir + "sample_batch_pairs.tsv", sep="\t")
+        if h.askif(
+            "Please upload the file ../sample_batch_pairs.tsv to the terra workspace as a new data table, and type yes once \
+      finished, else write no to quit and they will not be added"
+        ):
+            print("sample_batch_pair manually updated")
+
+    # replace string entries in sample_batch_pairs with references to sample_sets
+    myterra.updateReferences(wm, "sample_batch_pair", pair_df)
+
+    unique_pairs = wm.get_entities("sample_batch_pair").index.tolist()
+    sample_batch_pair_set_df = pd.DataFrame(
+        np.transpose(unique_pairs),
+        index=["all"] * len(unique_pairs),
+        columns=["sample_batch_pair"],
+    )
+    sample_batch_pair_set_df.index.name = "membership:sample_batch_pair_set_id"
+
+    # update sample_batch_pair_set to include all sample_batch_pairs
+    try:
+        wm.upload_entities("sample_batch_pair_set", sample_batch_pair_set_df)
+    except:
+        print("still can't update sample_batch_pair_set")
+        # in case it does not work
+        sample_batch_pair_set_df.to_csv(
+            working_dir + "sample_batch_pair_set.tsv", sep="\t"
+        )
+        if h.askif(
+            "Please upload the file ../sample_batch_pair_set.tsv to the terra workspace as a new data table, and type yes once \
+      finished, else write no to quit and they will not be added"
+        ):
+            print("sample_batch_pair_set manually updated")
+
+
+async def _CCLEFingerPrint(
```

Comment:
we should do like in the other tasks and have a generic function and a CCLE specific one.. that calls the generic + do the CCLE only things (like using taiga)